integrations/DCMFloodlight/utils.js

```diff
@@ -72,4 +91,224 @@ const mapFlagValue = (key, value) => {
+) => {
+  const randomNum = Math.random() * 10000000000000;
+  const customParams = {
+    ord: get(message, "properties.ord") || randomNum,
+    dc_lat: get(message, "context.device.adTrackingEnabled"),
+  };
+
+  if (salesTag) {
+    // sales tag
+    customParams.ord =
+      get(message, "properties.orderId") || get(message, "properties.order_id");
+    customParams.cost = get(message, "properties.revenue");
+    let qty = get(message, "properties.quantity");
+
+    // sums quantity from products array or fallback to properties.quantity
+    const products = get(message, "properties.products");
+    const quantities = calculateQuantity(products);
+    if (quantities) {
+      qty = quantities;
+    }
+
+    // Ref - https://support.google.com/campaignmanager/answer/2823450?hl=en#zippy=%2Chow-the-ord-parameter-is-displayed-for-each-counter-type%2Csales-activity-tags
+    switch (countingMethod) {
+      case "transactions":
+        customParams.qty = 1;
+        break;
+      case "items_sold":
+        if (qty) {
+          customParams.qty = parseFloat(qty);
+        }
+        break;
+      default:
+        logger.error("[DCM Floodlight] Sales Tag:: invalid counting method");
+        return;
+    }
+  } else {
+    // counter tag
+    // Ref - https://support.google.com/campaignmanager/answer/2823450?hl=en#zippy=%2Ccounter-activity-tags%2Chow-the-ord-parameter-is-displayed-for-each-counter-type
+    switch (countingMethod) {
+      case "standard":
+        break;
+      case "unique":
+        customParams.ord = 1;
+        customParams.num = get(message, "properties.num") || randomNum;
+        break;
+      case "per_session":
+        customParams.ord = get(message, "properties.sessionId");
+        if (!customParams.ord) {
+          logger.error(
+            "[DCM Floodlight] Counter Tag:: sessionId is missing from properties"
+          );
+          return;
+        }
+        break;
+      default:
+        logger.error("[DCM Floodlight] Counter Tag:: invalid counting method");
+        return;
+    }
+  }
+
+  // COPPA, GDPR, npa must be provided inside integration object
+  let { DCM_FLOODLIGHT } = message.integrations;
+  if (!DCM_FLOODLIGHT) {
+    ({ DCM_FLOODLIGHT } = integrationObj);
+  }
+
+  if (DCM_FLOODLIGHT) {
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.COPPA)) {
+      customParams.tag_for_child_directed_treatment = mapFlagValue(
+        "COPPA",
+        DCM_FLOODLIGHT.COPPA
+      );
+    }
+
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.GDPR)) {
+      customParams.tfua = mapFlagValue("GDPR", DCM_FLOODLIGHT.GDPR);
+    }
+
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.npa)) {
+      customParams.npa = mapFlagValue("npa", DCM_FLOODLIGHT.npa);
+    }
+  }
+
+  if (isDefinedAndNotNull(customParams.dc_lat)) {
+    customParams.dc_lat = mapFlagValue("dc_lat", customParams.dc_lat);
+  }
+
+  const matchId = get(message, "properties.matchId");
+  if (matchId) {
+    customParams.match_id = matchId;
+  }
+
```

Comment:
This piece of code is duplicate in both `buildIframeTrackPayload` and `buildGtagTrackPayload`

Reply:
Code Refactored. Created a separate function to handle this :- `buildCustomParamsUsingIntegrationsObject`